### PR TITLE
check `oversampling` in 1D

### PR DESCRIPTION
**Pull request by lostanlen**
in `check_runtime_args`

and write `test_check_runtime_args`

### CODE REVIEW

**Review by MuawizChaudhary — APPROVED** [review]
Should oversampling be set by user?

**Comment by lostanlen** [comment]
Yes. although i'll admit i don't love the current semantics. `alpha` also achieves a form of oversampling when getting the j's from xi's and sigma's. It's also a but surprising that the stride (`if average`) is not `T` but `T/2**oversampling`

But these are discussions for another day. here the goal is to make sure that the value of oversampling is OK at runtime.

**Comment by lostanlen** [comment]
i've written this to complement #876. The new asserts rely on the assumption that oversampling>=0, but this wasnt being checked at runtime until now

File: kymatio/scattering1d/frontend/base_frontend.py [hunk 1]
```diff
@@ -134,12 +134,17 @@ def output_size(self, detail=False):
...
         if not self.out_type in ('array', 'dict', 'list'):
-            raise RuntimeError("The out_type must be one of 'array', 'dict', or 'list'.")
+            raise ValueError("The out_type must be one of 'array', 'dict'"
+                             ", or 'list'. Got: {}".format(self.out_type))
 
         if not self.average and self.out_type == 'array':
             raise ValueError("Cannot convert to out_type='array' with "
                              "average=False. Please set out_type to 'dict' or 'list'.")
 
+        if self.oversampling<0:
```
Comment (cyrusvahidi) [inline kymatio/scattering1d/frontend/base_frontend.py:144]: change to `self.oversampling < 0`

File: tests/scattering1d/test_torch_scattering1d.py [hunk 2]
```diff
@@ -439,13 +439,12 @@ def test_Q(device, backend):
         _ = Scattering1D(
             J, shape, Q=0.9, backend=backend, frontend='torch')
     assert "Q should always be >= 1" in ve.value.args[0]
-
+
```
Comment (cyrusvahidi) [inline tests/scattering1d/test_torch_scattering1d.py:442]: remove?

Comment (lostanlen) [reply tests/scattering1d/test_torch_scattering1d.py]: my bad. my editor (Atom) added this automatically
6f3715b

**Review by MuawizChaudhary — APPROVED** [review]
Will merge

**Review by MuawizChaudhary — APPROVED** [review]

**Pull request merged**